californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java

```diff
@@ -61,21 +61,28 @@
 	private ScheduledFuture<?> statusLogger;
 	private ScheduledExecutorService scheduler;
 	private MessageIdProvider messageIdProvider;
+	private TokenProvider tokenProvider;
 	private SecureRandom secureRandom;
 
 	/**
 	 * Creates a new store for configuration values.
 	 * 
 	 * @param config the configuration to use.
+	 * @param tokenProvider the TokenProvider which provides CoAP tokens that
+	 *            are guaranteed to be not in use.
+	 * 
 	 */
-	public InMemoryMessageExchangeStore(final NetworkConfig config) {
+	public InMemoryMessageExchangeStore(final NetworkConfig config, TokenProvider tokenProvider) {
```

Comment:
I think we should still have a constructor that only takes a `NetworkConfig` as argument. We should also provide a setter for `TokenProvider` and during startup we can then check if a token provider is set and if not simply instantiate an `InMemoryRandomTokenProvider`. This way we do not need to change the test cases in so many places ....